### ISSUE TITLE
Error when ReadGenoVCF

### ISSUE THREAD

**Issue by alyazeeditalal** [gact/shmootl#2]
Dear shmootl developer, 

I am using the package to convert a vcf file to R/qtl cross geno object. The vcf file contains nematode SNPs information for 100 recombinant inbred lines (RILs) and 2 parents used to establish the RILs. I am using the following syntax to read the vcf file and returning it to R/qtl cross. 

library(shmootl)
rils <- getSamplesVCF("./Merged_RIAILs_SNPs/RIAILs_SNPs_filtered_position.vcf.recode.vcf")
rils
readGenoVCF("./Merged_RIAILs_SNPs/RIAILs_SNPs_filtered_position.vcf.recode.vcf", samples = rils, founders = mapping(c(APS7 = 'A', APS14 = 'B')))

However, it returns the following error message. 
Error in value[[3L]](cond) : 
  mapframe-like object has invalid sequence labels

I was very glad that I found this package that will save so much time when converting a vcf file to a format compatible to rqtl rather than doing it the usual way, especially when converting a large vcf file. It will indeed save me so much time and effort. 

Many Thanks, 
Talal

**Comment by tw164** [comment]
Dear Talal,

Thanks for your interest in using shmootl. I'll investigate this error and try to find a solution as soon as possible.

Regards,

Tom.

**Comment by alyazeeditalal** [comment]
Thanks, Tom. 

Talal

**Comment by tw164** [comment]
Hello again Talal,

The error you encountered was due to a chromosome ID that shmootl considered invalid. The chromosome ID was almost certainly fine, it's just that shmootl assumes that its input data relates to yeast samples, and is designed to raise an error if it finds something unexpected.

There is another package - https://github.com/gact/utl - that contains generic versions of some shmootl functions. I've added a function named `convert_vcf_to_genfile` to that package, and it should provide similar functionality to `readGenoVCF`, the main difference being that it writes the genotype data directly to a genotype file.

It should then be possible to load that genotype file along with its corresponding phenotype file using the **R/qtl** function `read.cross` with the `csvs` format parameter.

Let me know if the new function circumvents the issue.

Regards,

Tom.
